helpers/cahash/cahash.go

```diff
@@ -36,7 +36,7 @@ func GenerateHash(certRaw []byte) (string, error) {
 
 // Iterates over pem blocks until a non-CA certificate if found or no other PEM
 // blocks exist.
-func decodeOneCert(raw []byte) (*x509.Certificate, error) {
+func DecodeOneCert(raw []byte) (*x509.Certificate, error) {
```

Comment:
I would also use a guard close to keep the indentation cleaner, so a check for the byte length before the decoding:

```
if len(byteData) == 0 {
    return nil, errors.New("failed find PEM data")
}
// decode and parse
```

Then having a look at the [original func](https://github.com/paketo-buildpacks/ca-certificates/blob/main/cacerts/certs.go) it was expecting only one cert, while we are returning the first one.
I'm not sure about it but I guess it's fine (?).

Reply:
I can insert the guard, sure. I am not sure if that is better however, because then the loop will do the same check and fall through anyway.

Point is, the loop __cannot be eliminated__.

Your statement `we are returning the first one.` is not true as such.

Note the `if !cert.IsCA { ...` check in line 57. The code actually delivers the `first cert, which is not a CA cert`.

This desired cert may be deep into the PEM data, thus the loop.

Reply:
Yes, sorry, what I meant is that if the PEM contains multiple certs it will return the first "not-CA" one in the chain, but I'm not an experte of certs/ssl stuff, so I might be wrong here. :smile: